libraries/botframework-connector/src/auth/msalServiceClientCredentialsFactory.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ConfidentialClientApplication } from '@azure/msal-node';
+import { Configuration } from 'botbuilder-dialogs-adaptive-runtime-core';
+import { MsalAppCredentials } from './msalAppCredentials';
+import { ServiceClientCredentials } from '@azure/ms-rest-js';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+import { AuthenticationConstants } from './authenticationConstants';
+import { GovernmentConstants } from './governmentConstants';
+
+/**
+ * An implementation of ServiceClientCredentialsFactory that generates MsalAppCredentials
+ */
+export class MsalServiceClientCredentialsFactory implements ServiceClientCredentialsFactory {
+    private readonly appId: string;
+
+    /**
+     * Create an MsalServiceClientCredentialsFactory instance using runtime configuration and an
+     * @azure/msal-node ConfidentialClientApplication.
+     *
+     * @param configuration Runtime configuration.
+     * @param clientApplication An @azure/msal-node ConfidentialClientApplication instance.
+     */
+    constructor(configuration: Configuration, private readonly clientApplication: ConfidentialClientApplication) {
+        const appId = configuration.get(['MicrosoftAppId']) ?? '';
+        if (typeof appId !== 'string') {
+            throw new Error('Invalid appId');
+        }
+
+        this.appId = appId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async isValidAppId(appId: string): Promise<boolean> {
+        return appId === this.appId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async isAuthenticationDisabled(): Promise<boolean> {
+        return !this.appId;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async createCredentials(
+        appId: string,
+        audience: string,
+        loginEndpoint: string,
+        _validateAuthority: boolean
```

Comment:
Apparently MSAL for Node does not allow you to disable authority validation ([see here](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-node-migration#configure-msal)).